hls4ml/templates/vivado/nnet_utils/nnet_recurrent.h

```diff
@@ -0,0 +1,851 @@
+    nnet::tanh<data_T, typename CONFIG_T:: weight_t, typename CONFIG_T::ACT_CONFIG_T>(inputacc_c, tmpres_c);
+  }
+  // Operation: s=g*i+sold*f (update state with buffer to avoid timing issues)
+  for(int iacc = 0; iacc < (CONFIG_T::n_state); iacc++) {
+    #pragma HLS UNROLL
+    s_newstate[iacc] =  tmpres_c[iacc]*tmpres_ifo[iacc] + s_newstate[iacc]*tmpres_ifo[iacc+(CONFIG_T::n_state)];
+  }
+  // Operation: h=act(s)*o
+  if (CONFIG_T::ACT_CONFIG_T::activation_type == activ_relu){
+    nnet::relu<data_T, typename CONFIG_T:: weight_t, typename CONFIG_T::ACT_CONFIG_T>(s_newstate,s_actstate);
+  }
+  else if (CONFIG_T::ACT_CONFIG_T::activation_type == activ_sigmoid){
+    nnet::sigmoid<data_T, typename CONFIG_T:: weight_t, typename CONFIG_T::ACT_CONFIG_T>(s_newstate,s_actstate);
+  }
+  else if (CONFIG_T::ACT_CONFIG_T::activation_type == activ_tanh){
+    nnet::tanh<data_T, typename CONFIG_T:: weight_t, typename CONFIG_T::ACT_CONFIG_T>(s_newstate,s_actstate);
+  }
+  for(int iacc = 0; iacc < CONFIG_T::n_state; iacc++) {
+#pragma HLS UNROLL
+    h_state[iacc] = tmpres_ifo[iacc+2*(CONFIG_T::n_state)]*s_actstate[iacc];
+    h_newstate[iacc] = h_state[iacc];
+  }
+  # ifndef __SYNTHESIS__
+  std::cout << "Post-State: s [ "; for (int ii = 0; ii < CONFIG_T::n_state; ii++) std::cout << s_newstate[ii] << " "; std::cout << "]" << std::endl;
+  std::cout << "Post-State: h [ "; for (int ii = 0; ii < CONFIG_T::n_state; ii++) std::cout << h_newstate[ii] << " "; std::cout << "]" << std::endl;
+  # endif
+}
+
+template<class data_T, class res_T, typename CONFIG_T>
+  void lstm_loop(
+      data_T    data      [CONFIG_T::n_sequence*CONFIG_T::n_in],
+      res_T     res [CONFIG_T::n_sequence_out*CONFIG_T::n_state],
+      typename CONFIG_T::weight_t     param  [CONFIG_T::n_state*4*CONFIG_T::n_in],
+      typename CONFIG_T::weight_t     param_r[CONFIG_T::n_state*4*CONFIG_T::n_state],
+      typename CONFIG_T::bias_t     param_b[CONFIG_T::n_state*4],
+      typename CONFIG_T::bias_t     param_br[CONFIG_T::n_state*4]
+      ) {
+
+    res_T     h_newstate[CONFIG_T::n_state];
+    data_T data_in[CONFIG_T::n_in];
+    bool reset_state = true;
+
+    #pragma HLS ARRAY_PARTITION variable=h_newstate complete
+    for(int ii = 0; ii < CONFIG_T::n_state; ii++) h_newstate[ii] = 0;
+    for(int iloop = 0; iloop < CONFIG_T::n_sequence; iloop++) {
+      for(int j = 0; j < CONFIG_T::n_in; j++){data_in[j] =  data[j + iloop*CONFIG_T::n_in];}
+      nnet::lstm_static<data_T, res_T, CONFIG_T>(reset_state,data_in,h_newstate, param,param_r,param_b, param_br);
+      if (CONFIG_T::n_sequence_out > 1)
+        for(int i=CONFIG_T::n_state*iloop, j=0; i<(CONFIG_T::n_state*(iloop+1)); i++,j++){
+          res[i] = h_newstate[j];
+      }
+      reset_state = false;
+    }
+    if (CONFIG_T::n_sequence_out == 1)
+      for(int i=0; i<(CONFIG_T::n_state); i++){
+        res[i] = h_newstate[i];
+    }
+}
+
+template<class data_T, class res_T, typename CONFIG_T>
+  void lstm_loop(
```

Comment:
I would prefer if this was called just `lstm` or `lstm_stack` and then the loop iteration should be `lstm_loop`

Reply:
changed to lstm_stack